### ISSUE TITLE
Configuration 1 and 4

### ISSUE THREAD

**Issue by torayeff** [SmartDataAnalytics/MA-INF-4222-NLP-Lab#73]
I can not understand the configuration 1 for the exercises 2. 

You say that we need to report these performance measures:

model a - train - [performance measures][0:4] 
model a - test - [performance measures][0:4]

and we need to use kfold cross validation. It is okay with performance measures for the training set. I will split the whole dataset into training and test sets (0.75/0.25). Then I will do cross validation on the training set, for this I can report 5 measurements for each fold. 

But how can I report 5 measurements for the test set? I mean why should I do cross validation again on the test set after learning the model with the split training set.

I am bit confused with this configuration, others are okay. Can you please clarify me this?

**Comment by diegoesteves** [comment]
Hi @torayeff basically just call the method *train_test_split* K-times defining the random state variable.

Please see an example above:

```
...
for d in range(len(r)):
   Xtr, Xte, ytr, yte = train_test_split(X, y, test_size=0.25, random_state=r[d])
   m = cls.fit(Xtr, ytr)
   ypr = m.predict(Xte)
   print(metrics.flat_classification_report(yte, ypr, labels=sorted_labels.keys(), target_names=sorted_labels.values(), digits=3))
...
```    
Another possibility is to use the K-fold method:
```
kf = KFold(n_splits=5, random_state=r) 
kf.get_n_splits(X)
for train_index, test_index in kf.split(X):
...
```
You helped me to notice that I have not defined the set of random states (just 4222 will obviously lead to the same split over and over, which is senseless). So please consider the set **r=[42, 39, 10, 5, 50]** in this case.

**Comment by torayeff** [comment]
Thanks @diegoesteves for example. I would just add one way to do the cross validation in scipy:
```
cv_info = cross_validate(model_a, X, y, cv=5, return_train_score=True, 
               scoring=['precision_macro', 'recall_macro', 'accuracy', 'f1_macro'])
pprint.pprint(cv_info)
```

**Comment by diegoesteves** [comment]
Hi, there was a mistake from my side: in the script I created for generating the dataset3 I did not consider sampling (actually was more expecting that this would serve as a goldstandard so that people could apply K-fold on it afterward, but it seems I was not clear enough). So I will consider performance measures for configuration 4 as it is, i.e., just train and test, no sampling. 
Thanks for pointing out.
